src/main/java/christmas/model/order/OrderItems.java

```diff
@@ -0,0 +1,67 @@
+package christmas.model.order;
+
+import christmas.model.menu.MenuType;
+import java.time.LocalDate;
+import java.util.List;
+
+public class OrderItems {
+
+    private List<OrderItem> orderItems;
+
+    public OrderItems(List<OrderItem> orderItems) {
+        validate(orderItems);
+        this.orderItems = orderItems;
+    }
+
+    private void validate(List<OrderItem> orderItems) {
+        validateOrderQuantity(orderItems);
+        validateDuplication(orderItems);
+        validateOnlyDrinkMenu(orderItems);
+    }
+
+    private void validateOrderQuantity(List<OrderItem> orderItems) {
+        long totalQuantity = orderItems.stream()
+                .mapToInt(OrderItem::getQuantity)
+                .sum();
+        if (totalQuantity > 20) {
+            throw new IllegalArgumentException("[ERROR] 유효하지 않은 주문입니다. 다시 입력해 주세요.");
+        }
+    }
+
+    private void validateDuplication(List<OrderItem> orderItems) {
+        boolean hasDuplicationOrderItem = orderItems.stream()
+                .distinct()
+                .toList()
+                .size() != orderItems.size();
+        if (hasDuplicationOrderItem) {
+            throw new IllegalArgumentException("[ERROR] 유효하지 않은 주문입니다. 다시 입력해 주세요.");
+        }
```

Comment:
(이런 방법도 있어요!)

```suggestion
        if(new HashSet<>(orderItems).size() != orderItems.size()){
            throw new IllegalArgumentException();
        }
```
간단한 중복검사는 set 자료구조를 통해서 쉽게 코드를 작성할 수 있습니다 :)

Reply:
set을 쓸까하다가 stream을 써봤네요 추천 감사합니다